test/units/testsuite-35.sh

```diff
@@ -360,6 +360,89 @@ EOF
     fi
 }
 
+setup_idle_action_lock() {
+    useradd testuser ||:
+
+    mkdir -p /run/systemd/logind.conf.d/
+    cat >/run/systemd/logind.conf.d/idle-action-lock.conf <<EOF
+[Login]
+IdleAction=lock
+IdleActionSec=1s
+EOF
+    systemctl restart systemd-logind.service
+    systemctl service-log-level systemd-logind.service debug
+}
+
+teardown_idle_action_lock() {
+    set +e
+    rm -f /run/systemd/logind.conf.d/idle-action-lock.conf
+    pkill -9 -u "$(id -u testuser)"
+    userdel -r testuser
+    systemctl restart systemd-logind.service
+}
+
+test_lock_idle_action() {
+    if ! command -v expect >/dev/null ; then
+        echo >&2 "expect not installed, skiping test ${FUNCNAME[0]}"
+        return 0
+    fi
+
+    setup_idle_action_lock
+    trap teardown_idle_action_lock RETURN
+
+    if loginctl --no-legend | awk '{ print $3; }' | sort -u | grep -q testuser ; then
+        echo >&2 "Session of the \'testuser\' is already present."
+        return 1
+    fi
+
+    # IdleActionSec is set 1s but the accuracy of associated timer is 30s so we
+    # need to sleep in worst case for 31s to make sure timer elapsed. We sleep
+    # here for 35s to accomodate for any possible scheudling delays.
+    cat > /tmp/test.exp <<EOF
+spawn systemd-run -G -t -p PAMName=login -p User=testuser bash
```

Comment:
Not sure, but why not to simply run a script with `systemd-run`??

Reply:
I am using expect because I want tty allocated and AFAICT you can't do systemd-run --pty from context where you don't have any tty.